eng/native/init-vs-env.cmd

```diff
@@ -6,9 +6,8 @@
 set "__VCBuildArch="
 if /i "%~1" == "x86"   (set __VCBuildArch=x86)
 if /i "%~1" == "x64"   (set __VCBuildArch=x86_amd64)
```

Comment:
Is cross-compiling arm64 (target) on x64 or x86 (host) possible with the toolchain?

Reply:
Judging by my the bits own my installation, it seems like it is

![image](https://github.com/dotnet/runtime/assets/2813363/0f0c9f87-1502-4e6d-8c5f-2f890bdd58cc)

Reply:
I'd be inclined to not try to fix arm64 to x64 cross compiling in this PR though.

Reply:
Isn't that just about stamping the same pattern on the x86 and x64 lines?

Reply:
I would think so. My only concern is that nobody uses that scenario at the moment, so other than me testing locally today there's nothing exercising that path.

Reply:
Right, it is a favor for the next person with arm64 machine needing to build in this config.

Reply:
I'll see what I can do.

Reply:
I made an attempt at addressing cross-compilation in general as described in https://github.com/dotnet/runtime/issues/76516 but I hit a number of issues in various projects. For example, the `gen-buildsys.cmd` script is being passed the host architecture but not the target architecture

https://github.com/dotnet/runtime/blob/2fe4afa81e5986124a1fef91e6f35f8994b92ef7/src/coreclr/build-runtime.cmd#L389

https://github.com/dotnet/runtime/blob/2fe4afa81e5986124a1fef91e6f35f8994b92ef7/eng/native/gen-buildsys.cmd#L21

which is causing build errors downstream. Ultimately, I find that the "host architecture" parameter is problematic because it simply reflects the process architecture of the build entrypoint and it is occasionally conflated with the target architecture. An eventual solution should simply remove that parameter and replace it with the OS architecture, since I suspect the reasons that introduced it in the first place are obsolete.

With that said, I'm going to merge this PR as-is to unblock native arm64 builds and file an issue tracking ARM64 cross compilation support.